app/controllers/reports/regions_controller.rb

```diff
@@ -10,8 +10,9 @@ def index
   end
 
   def show
-    @region = Facility.find_by!(slug: facility_params[:id])
+    @region = scope.find_by!(slug: facility_params[:id])
     authorize(:dashboard, :show?)
+    RequestStore.store[:force_cache] = true if force_cache?
```

Comment:
Actually, now that I look at this, it seems like there's literally only one layer where you'd need to pass `force_cache: true`, which is from the controller to the service. Not sure if `RequestStore` is really that valuable here, since you also need to manage setting and reading it anyway.

I imagine, defaulting to `cache:true` and setting it to false when the param is set is nearly as much code and probably simpler?

Reply:
The problem with passing the param down is that you then need to pass it into all the dependant services. That results in one more param for every initialize call for something that really is a cross-cutting concern, and not directly related to the data we are retrieving. 

If you look at an earlier commit - https://github.com/simpledotorg/simple-server/pull/1150/commits/2e260c819610854b01eb136857aaaa083dd0d808 - you'll see the ugliness of passing around storing all that additional instance state, as opposed to using RequestStore.

Reply:
Ah, that makes sense, the diff didn't capture the ugly enough.

I guess the only thing to be wary of is using these service classes is to remember to globally and appropriately set the `RequestStore.store[:force_cache]` wherever it is called (non-controller usage, tests etc.) when we want a cached response?

Reply:
> I guess the only thing to be wary of using these service classes is to remember to globally and appropriately set the `RequestStore.store[:force_cache]` wherever it is called (non-controller usage, tests etc.) when we want a cached response?

Yup, I imagine this will primarily be used for busting the cache from the controller layer.  I can't imagine where we will often want to use it for non-web driven purposes, but it is there if we need it.

Reply:
Oh, and to be clear, the default behavior should always be to fetch from the cache if `force_cache` isn't set...`force_cache` is only ever intended to mean "I know better than the caching mechanism, grab fresh data and populate the cache with it".